properties/base/base.py

```diff
@@ -22,6 +22,42 @@
 GENERIC_ERRORS = (ValueError, KeyError, TypeError, AttributeError)
 
 
+def build_from_bases(bases, classdict, attr, attr_dict):
+    """Helper function to build private HasProperties attributes"""
+    output = OrderedDict()
+    output_keys = []
+    all_bases = []
+    # Go through the bases from furthest to nearest ancestor
+    for base in reversed(bases):
+        # Only keep the items that are still defined on the bases
+        if base is not object and issubclass(base, HasProperties):
+            output_keys += list(getattr(base, attr).keys())
+        # Collect all bases so we ensure overridden items are assigned
+        # in the correct order
+        for item in reversed(base.__mro__):
+            if item is object or not issubclass(item, HasProperties):
+                continue
+            if item not in all_bases:
+                all_bases.append(item)
+    # Update the items in reverse MRO order; only keep those that are
+    # defined on the bases
+    for base in all_bases:
+        for key, val in iteritems(getattr(base, attr)):
+            if key in base.__dict__ and key in output_keys:
+                output.update({key: val})
```

Comment:
Those won't be in the `_prop` dict though. So this is the reason we have `output_keys` - i.e. if we are building this class:
```python
class AnotherSubClass(SubClass):
    pass
```
it will determine `output_keys` from just the direct bases. Since `a` was overridden there with a non-`property`, it will not pick up the `property` from the class further up the inheritance tree...

Reply:
Cool